pkg/docker/config/config.go

```diff
@@ -587,6 +607,16 @@ func findAuthentication(registry, path string, legacyFormat bool) (types.DockerA
 		return getAuthFromCredHelper(ch, registry)
 	}
 
+	// Support for different paths in auth.
+	if ref != nil {
```

Comment:
Just to be sure, restrict this to `!legacyFormat` where URLs may exist?

Reply:
Sounds good, changed as suggested.